### PR TITLE
Added feature to trigger retry on a PR

### PR DESCRIPTION
**Pull request by eidelmanjonathan**
Added feature to trigger retry on a PR to rerun proofs when there is an intermittent issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ci/snapshot/prepare_source.py [hunk 1]
```diff
@@ -383,7 +383,7 @@ def make_tarfile_name(repository, sha=None):
     filename += '-{:04}{:02}{:02}-{:02}{:02}{:02}'.format(
         now.year, now.month, now.day, now.hour, now.minute, now.second)
     if sha:
-        filename += '-{}'.format(sha.lower())
+        filename += '-{}'.format(sha.replace("/", "_").lower())
```
Comment (karkhaz) [inline ci/snapshot/prepare_source.py:386]: Not sure why this is needed for this feature?

File: ci/snapshot/cbmc_ci_start.py [hunk 2]
```diff
@@ -95,6 +95,7 @@ def lambda_handler(event, context):
             "http",
             "http_dev",
             "release-candidate",
+            "COMMENT_RETRY"
```
Comment (karkhaz) [inline ci/snapshot/cbmc_ci_start.py:98]: How does this work? I thought that we check out the branch that we want to check. If you're setting the branch to `COMMENT_RETRY` above, how does checking the branch out not fail?

File: ci/snapshot/prepare_source.py (hunk 1 above)
Comment (karkhaz) [reply ci/snapshot/prepare_source.py:386]: Oh, never mind, just read the commit message.

File: ci/snapshot/cbmc_ci_github.py [hunk 3]
```diff
@@ -225,6 +229,32 @@ def parse_pr(body):
...
+    Parse the pull request event body for the base and head branches,
+    and get the repository name, repository id, and branch name for the base
+    branch, and get the sha for the head commit on the head branch.
+
+    event has a format as described here:
+        https://developer.github.com/v3/activity/events/types/#pullrequestevent
+
+    """
+    comment_text = body["comment"]["body"]
+    if comment_text in ["CBMC_RETRY"]:
```
[the remark this answers is not in the document]
Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_github.py:243]: Done

File: ci/snapshot/cbmc_ci_github.py [hunk 4]
```diff
@@ -225,6 +229,32 @@ def parse_pr(body):
...
+    """
+    Parse the pull request event body for the base and head branches,
+    and get the repository name, repository id, and branch name for the base
+    branch, and get the sha for the head commit on the head branch.
+
+    event has a format as described here:
+        https://developer.github.com/v3/activity/events/types/#pullrequestevent
+
+    """
+    comment_text = body["comment"]["body"]
```
[the remark this answers is not in the document]
Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_github.py:242]: Done

File: ci/snapshot/prepare_source.py (hunk 1 above)
Comment (eidelmanjonathan) [reply ci/snapshot/prepare_source.py:386]: Right the issue is the tarfile can't have slashes in the name

File: ci/snapshot/cbmc_ci_start.py (hunk 2 above)
Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_start.py:98]: So we don't actually check out the branch, we check out the PR. The issue here is that we don't get the branch in the webhook data for a comment, so I just set it to some constant COMMENT_RETRY.

The code here is to hard code the branches we want to check (we probably should not hard code this but that's for a different PR), so I'm adding in that if there's this dummy retry branch we should check that too

Comment (eidelmanjonathan) [reply ci/snapshot/cbmc_ci_start.py:98]: Added a comment

**Pull request merged**